package/MDAnalysis/coordinates/base.py

```diff
@@ -1282,8 +1308,85 @@ def __repr__(self):
+        """Add auxiliary data to be read alongside trajectory.
+
+        Auxiliary data may be any data timeseries from the trajectory additional
+        to that read in by the trajectory reader. *auxdata* can be an AuxReader 
+        instance, or the data itself as e.g. an array, filename; in the latter 
+        case an appropriate AuxReader is guessed from the data/file format.
+
+        The representative value of the auxiliary data for each timestep (as
+        calculated by the AuxReader) is stored in the ts.aux namespace as 
+        auxname.
+
+        e.g. to add additional pull force data stored in pull-force.xvg::
 
+          u = MDAnalysis.Universe(PDB, XTC)
+          u.trajectory.add_auxiliary('pull', 'pull-force.xvg')
 
+        The representative value for the current timestep may then be accessed 
+        as ``u.trajectory.ts.aux.pull``.
+
+        AuxReader is initially matched to the current timestep of the trajectory.
+
+        """
+        if auxname in self.aux_list:
+            raise ValueError("Auxiliary data with name {name} already "
+                             "exists".format(name=auxname))
+        if isinstance(auxdata, AuxReader):
+            aux = auxdata
+        else:
+            auxreader = get_auxreader_for(auxdata)
+            aux = auxreader(auxname, auxdata, **kwargs)
+        self._auxs[auxname] = aux
+        self.ts = aux.go_to_ts(self.ts)
+    
+    def remove_auxiliary(self, auxname):
+        """Close the Reader for auxiliary data *auxname* and remove data 
+        from trajectory."""
+        if auxname in self.aux_list:
+            self._auxs[auxname].close()            
+            del self._auxs[auxname]
+            del self.ts.aux.__dict__[auxname]
+        else:
+            raise ValueError("No auxiliary named {name}".format(name=auxname))
+            
+    @property
+    def aux_list(self):
+        """ Lists the names of added auxiliary data. """
+        return self._auxs.keys()
+
+    def next_as_aux(self, auxname):
+        """ Move to the next trajcetory timestep which has a step from the
+        auxiliary *auxname* assigned to it.
+        If the auxiliary steps are less frequent (aux dt > trajectroy dt), 
+        this allows to move forward at the auxiliary pace (rounded to nearest
+        timestep).
+        If the auxiliary steps are more frequent than the trajectory 
+        timesteps (aux dt < trajectory dt), this works the same as just 
+        calling next() on the trajectory."""
+        aux = self._auxs[auxname]
+        ts = self.ts
+        if aux.step > aux.n_steps-1:
+            raise StopIteration
+        next_frame = aux.step_to_frame(aux.step+1, ts)
+        while self.frame != next_frame:
+            ts = self.next()
+        return ts 
+
+    def iter_as_aux(self, auxname):
+        """Iterate through trajectory steps which have a step from the
+        auxiliary *auxname* assigned to the."""
+        self._reopen()
```

Comment:
I think it has to be reopen, so it starts before the first frame (like in __iter__); rewind takes it to the first frame